docs/root/intro/arch_overview/advanced/data_sharing_between_filters.rst

```diff
@@ -88,3 +89,31 @@ the class definition (e.g., HTTP protocol, requested server name, etc.). In
 addition, it provides a facility to store typed objects in a map
 (``map<string, FilterState::Object>``). The state stored per filter can be
 either write-once (immutable), or write-many (mutable).
+
+.. _arch_overview_advanced_filter_state_sharing:
+
+Filter state sharing
+--------------------
+
+Filter state objects are bound to the lifespan of the associated parent stream.
+However, by marking a downstream object as shared with the upstream connection
+during creation, the object is shared with the upstream connection filter
+state, and its lifespan is extended beyond the original stream. Any upstream
+TCP or HTTP filter can access the shared object. Upstream transport sockets can
+also read the shared objects and customize the creation of the upstream
+transport. For example, the :ref:`internal upstream transport socket
+<envoy_v3_api_msg_extensions.transport_sockets.internal_upstream.v3.InternalUpstreamTransport>`
+copies references to the shared objects to the internal connection downstream
+filter state.
+
+The filter state objects that are shared with the upstream also affect the
+connection pooling decisions if they implement a hashing interface. Whenever a
+shared hashable object is added, an upstream connection is created for each
+distinct hash value, which ensures that these objects are not overwritten by
+subsequent downstream requests to the same upstream connection. For example, a
+custom HTTP filter may create a shared hashable object from the value of a
+special header. In this case, a separate upstream connection is created for
```

Comment:
Is there any monitoring: stat, access log, etc. that would allow someone to know that they're creating all of these connection pools?

Reply:
How does connection count allow you to distinguish it's b/c the upstream has significant traffic vs b/c we have many pools?

Reply:
seems like a stat here for number of pools makes sense, not blocking for this, can be done as a follow up